File: Persona.php
<?php
/**
 *
 */
 include 'Conexion.php';
class Persona
{

  private $_idPersona;
  private $_nombre;
  private $_apellidoP;
  private $_apellidoM;
  private $_habilidades;
  private $_algoMas;

  private $_objConexion;

  function __construct()
  {
    $this->_objConexion=new Conexion();
    $this->_objConexion->Conectar();
  }

  public function getAllPersonas()
  {
    $_personas = ($this->_objConexion->traerRegistros("SELECT * FROM personas"));
    $_lista="";
    foreach ($_personas as $key) {
      $_lista.='{';
      $_lista.='"idPersona":"'.$key['idPersona'].'",';
      $_lista.='"nombre":"'.$key['nombre'].'",';
      $_lista.='"apellidoP":"'.$key['apellidoP'].'",';
      $_lista.='"apellidoM":"'.$key['apellidoM'].'",';
      $_lista.='"habilidades":"'.$key['habilidades'].'",';
      $_lista.='"algoMas":"'.$key['algoMas'].'"';
      $_lista.='},';
    }
    $_lista = trim($_lista, ',');
    return '{"data":['.$_lista.']}';
  }

  public function newPersona()
  {
    $this->_idPersona = $this->_objConexion->getNextID("personas","idPersona");
    return ($this->_objConexion->IEA("INSERT INTO personas VALUES ('$this->_idPersona', '$this->_nombre', ' $this->_apellidoP', '$this->_apellidoM', '$this->_habilidades', '$this->_algoMas')"));
  }

  public function updatePersona()
  {
    return ($this->_objConexion->IEA("UPDATE personas SET nombre = '$this->_nombre', apellidoP = ' $this->_apellidoP', apellidoM = '$this->_apellidoM', habilidades = '$this->_habilidades', algoMas = '$this->_algoMas' WHERE idPersona = $this->_idPersona;"));
  }

  public function deletePersona()
  {
    return ($this->_objConexion->IEA("DELETE FROM personas WHERE idPersona = $this->_idPersona"));
  }



    /**
     * Get the value of Id Persona
     *
     * @return mixed
     */
    public function getIdPersona()
    {
        return $this->_idPersona;
    }

    /**
     * Set the value of Id Persona
     *
     * @param mixed $_idPersona
     *
     * @return self
     */
    public function setIdPersona($_idPersona)
    {
        $this->_idPersona = $_idPersona;

        return $this;
    }

    /**
     * Get the value of Nombre
     *
     * @return mixed
     */
    public function getNombre()
    {
        return $this->_nombre;
    }

    /**
     * Set the value of Nombre
     *
     * @param mixed $_nombre
     *
     * @return self
     */
    public function setNombre($_nombre)
    {
        $this->_nombre = $_nombre;

        return $this;
    }

    /**
     * Get the value of Apellido
     *
     * @return mixed
     */
    public function getApellidoP()
    {
        return $this->_apellidoP;
    }

    /**
     * Set the value of Apellido
     *
     * @param mixed $_apellidoP
     *
     * @return self
     */
    public function setApellidoP($_apellidoP)
    {
        $this->_apellidoP = $_apellidoP;

        return $this;
    }

    /**
     * Get the value of Apellido
     *
     * @return mixed
     */
    public function getApellidoM()
    {
        return $this->_apellidoM;
    }

    /**
     * Set the value of Apellido
     *
     * @param mixed $_apellidoM
     *
     * @return self
     */
    public function setApellidoM($_apellidoM)
    {
        $this->_apellidoM = $_apellidoM;

        return $this;
    }

    /**
     * Get the value of Habilidades
     *
     * @return mixed
     */
    public function getHabilidades()
    {
        return $this->_habilidades;
    }

    /**
     * Set the value of Habilidades
     *
     * @param mixed $_habilidades
     *
     * @return self
     */
    public function setHabilidades($_habilidades)
    {
        $this->_habilidades = $_habilidades;

        return $this;
    }

    /**
     * Get the value of Algo Mas
     *
     * @return mixed
     */
    public function getAlgoMas()
    {
        return $this->_algoMas;
    }

    /**
     * Set the value of Algo Mas
     *
     * @param mixed $_algoMas
     *
     * @return self
     */
    public function setAlgoMas($_algoMas)
    {
        $this->_algoMas = $_algoMas;

        return $this;
    }

}

 ?>

File: src/java/clases/Persona.java
package clases;

import conexion.Conexion;
import java.sql.ResultSet;
import java.sql.SQLException;

 public class Persona {

    private int idPersona;
    private String nombre;
    private String apellidoP;
    private String apellidoM;
    private String habilidades;
    private String algoMas;

    private Conexion conexion;

    public Persona() {
        conexion = new Conexion();
        conexion.Conectar();
    }

    public String getAllPersonas() {
        String lista = "";
        ResultSet listaPersonas = conexion.ejecutarConsulta("SELECT * FROM personas");
        try {
            while (listaPersonas.next()) {
                lista += "{";
                lista += "\"idPersona\":\"" + listaPersonas.getString("idPersona") + "\",";
                lista += "\"nombre\":\"" + listaPersonas.getString("nombre") + "\",";
                lista += "\"apellidoP\":\"" + listaPersonas.getString("apellidoP") + "\",";
                lista += "\"apellidoM\":\"" + listaPersonas.getString("apellidoM") + "\",";
                lista += "\"habilidades\":\"" + listaPersonas.getString("habilidades") + "\",";
                lista += "\"algoMas\":\"" + listaPersonas.getString("algoMas") + "\"";
                lista += "},";
            }     
            try{
                lista = lista.substring(0, lista.length() - 1);
            }catch(java.lang.StringIndexOutOfBoundsException err){
                err.printStackTrace();
            }
            System.out.println(lista);
        } catch (SQLException ex) {
            ex.printStackTrace();
        }
        return "{\"data\":[" + lista + "]}";
    }
    
    
  public int newPersona()
  {
    this.idPersona = this.conexion.getNextID("personas","idPersona");
    return (this.conexion.ejecutarRUD("INSERT INTO personas VALUES ('"+this.idPersona+"', '"+this.nombre+"', '"+this.apellidoP+"', '"+this.apellidoM+"', '"+this.habilidades+"', '"+this.algoMas+"')"));
  }

  public int updatePersona()
  {
    return (conexion.ejecutarRUD("UPDATE personas SET nombre = '"+this.nombre+"', apellidoP = '"+this.apellidoP+"', apellidoM = '"+this.apellidoM+"', habilidades = '"+this.habilidades+"', algoMas = '"+this.algoMas+"' WHERE idPersona = "+this.idPersona+";"));
  }

  public int deletePersona()
  {
    return (conexion.ejecutarRUD("DELETE FROM personas WHERE idPersona = "+this.idPersona));
  }
  
    public static void main(String[] args) {
        Persona p=new Persona();
        System.out.println(p.getAllPersonas());
        
    }

    public int getIdPersona() {
        return idPersona;
    }

    public void setIdPersona(int idPersona) {
        this.idPersona = idPersona;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellidoP() {
        return apellidoP;
    }

    public void setApellidoP(String apellidoP) {
        this.apellidoP = apellidoP;
    }

    public String getApellidoM() {
        return apellidoM;
    }

    public void setApellidoM(String apellidoM) {
        this.apellidoM = apellidoM;
    }

    public String getHabilidades() {
        return habilidades;
    }

    public void setHabilidades(String habilidades) {
        this.habilidades = habilidades;
    }

    public String getAlgoMas() {
        return algoMas;
    }

    public void setAlgoMas(String algoMas) {
        this.algoMas = algoMas;
    }

}
